### DISCUSSION TITLE
How to dismiss || ignore || delete push notifications?

### DISCUSSION THREAD

**Discussion by oikkoikk** [invertase/react-native-firebase#5287] [Q&A] [answered]
I want to filter some sort of push notifications _without using 'Topic'_ (because using 'Topic' + 'Token' both is unaccepted).
How can I dismiss the incoming push notifications( dismiss means totally ignore the notification)?

++ another question
How can I display the pop-up in the background and quit state on Android?

**Comment by andersonaddo** [comment, marked as answer]
Read this doc _entirely_:
https://rnfirebase.io/messaging/usage#receiving-messages

You're going to have to be a bit more specific about the type of notifications and fcm messages you're sending to the client.

> I want to filter some sort of push notifications without using 'Topic' (because using 'Topic' + 'Token' both is unexcepted).
> How can I dismiss the incoming push notifications( dismiss means totally ignore the notification)?

If you're sending non data-only fcm messages, then filtering will be difficult since its automatically shown via native side code.
If youre sending data-only messaging, then its easy to decide when to and not to show the notification since all that logic will be in your fcm handlers.

> How can I display the pop-up in the background and quit state on Android?

This should already be the case automatically for non data-only fcm notifications.
For data-only messages, that's handled like so:
https://rnfirebase.io/messaging/usage#background-application-state
